src/pages/library/[id].tsx

```diff
@@ -0,0 +1,168 @@
+/* eslint-disable react/jsx-one-expression-per-line */
+/* eslint-disable react/jsx-indent-props */
+/* eslint-disable react/jsx-indent */
+import { IoMdArrowBack } from "react-icons/io";
+
+import {
+    Avatar,
+    Box,
+    Button,
+    Flex,
+    IconButton,
+    Stack,
+    Tag,
+    Text,
+    useToast,
+} from "@chakra-ui/react";
+import { doc, getDoc } from "firebase/firestore";
+import { GetServerSideProps } from "next";
+import { useSession } from "next-auth/react";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import { ParsedUrlQuery } from "querystring";
+
+import { BookProps } from "../../interfaces/Book";
+import { UserProps } from "../../interfaces/User";
+import { db } from "../../services/firebase";
+import { borrowBook } from "../../store/books/actions";
+
+interface Props {
+    book: BookProps;
+}
+
+export default function Library({ book }: Props) {
+    const { data: session } = useSession();
+    const router = useRouter();
+    const toast = useToast();
+
+    const handleBorrowBook = async () => {
+        const user: UserProps = {
+            id: String(session?.user?.id),
+            name: String(session?.user?.name),
+            email: String(session?.user?.email),
+            image: String(session?.user?.image),
+            borrowedBooks: null,
+        };
+        await borrowBook(book, user);
```

Comment:
por que não passar loco session.user?
se for porque user pode ser undefined basta checar previamente:
```
if (!!session?.user){
  await borrowBook(book, session.user);
}
```